llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/utils.py

```diff
@@ -132,11 +140,16 @@ async def _completion_with_retry(**kwargs: Any) -> Any:
             tools = to_gemini_tools(tools) if tools else []
             generation_config = kwargs if kwargs else {}
             return await generation.send_message_async(
-                prompt, tools=tools, generation_config=generation_config
+                prompt,
+                tools=tools,
+                generation_config=generation_config,
+                safety_settings=client._safety_settings,
             )
         elif chat:
             generation = client.start_chat(**params)
-            return await generation.send_message_async(prompt, **kwargs)
+            return await generation.send_message_async(
+                prompt, safety_settings=client._safety_settings, **kwargs
```

Comment:
Is this the right way to pass this? Feels weird to access hidden variables on the client object. Especially since we are already calling `client.start_chat` 

How do google docs recommend doing this? Should be be storing the safety settings on the `Gemini` llm class and pass them into these functions instead?

Reply:
Done